src/server.cpp

```diff
@@ -37,44 +35,45 @@ void Server::handle_receive(const error_code& error, std::size_t bytes)
         return;
     }
 
-    size_t length = recv_buffer_[2] * 256 + recv_buffer_[3];
+    try
+    {
+        Packet packet = makeResponse(Packet(m_recvBuffer, bytes));
+
+        m_socket.async_send_to(boost::asio::buffer(packet.makeSendBuffer("secret"), 20),
```

Comment:
You don't need `20` here. The vector returned from `makeSendBuf` must be of correct size and `boost::asio::buffer` can deduce its size.